### ISSUE TITLE
[BUG] Salesforce Account Connection

### ISSUE THREAD

**Issue by aturow** [PipedreamHQ/pipedream#4230]
**Describe the bug**
An existing account to a Salesforce sandbox is disconnected i.e. red. When I try to reconnect, the Account is not updated but a new one is created. 

**To Reproduce**
1. Go to my account andrzejturowski
2. Click on try to reconnect salesforce test
3. See newly created connection not reconnecting the existing account.

**Expected behavior**
Expect to connect existing account.

**Screenshots**

<img width="772" alt="image" src="https://user-images.githubusercontent.com/113529148/190171915-4fa5e69c-9863-4c7f-b67a-3e2016226317.png">
<img width="708" alt="image" src="https://user-images.githubusercontent.com/113529148/190169944-b07714fc-7a08-498e-a389-1b541c2f428c.png">
<img width="598" alt="image" src="https://user-images.githubusercontent.com/113529148/190173617-d3d33d7f-730a-43c7-9895-cc5a819e8818.png">

<img width="787" alt="image" src="https://user-images.githubusercontent.com/113529148/190171707-bffc964d-6ddc-4376-9848-61a23b0b794b.png">

**Comment by DilanAthukorala** [comment]
@aturow Thanks for reporting this issue, Our team will look into this and get back to you. 

I have observed that the Authentication base url called from pipedream is invalid. 

expected base url:- **https://pipedream-b-dev-ed.develop.lightning.force.com**
Actual base url called by pipedream:- **https://pipedream-b-dev-ed.my.salesforce.com**

**Comment by dannyroosevelt** [comment]
> 

@DilanAthukorala can you tell me more about the authentication base URL being invalid?

**Comment by DilanAthukorala** [comment]
> > 
> 
> @DilanAthukorala can you tell me more about the authentication base URL being invalid?

@dannyroosevelt I was unable to connect to the authentication base url called from pipedream (looks like it doesn't exist) and it's different from the base URL of my actual Salesforce account.

**Comment by aturow** [comment]
Thank you for looking into this. With this new information it is
probably the new URL's that salesforce rolled out to have the word sandbox
in the url.
I will just update the workflows to use the new connection.
Best regards,
Andrzej

**Comment by dannyroosevelt** [comment]
> > > 
> > 
> > 
> > @DilanAthukorala can you tell me more about the authentication base URL being invalid?
> 
> @dannyroosevelt I was unable to connect to the authentication base url called from pipedream (looks like it doesn't exist) and it's different from the base URL of my actual Salesforce account.

@DilanAthukorala the Authorization, Access Token, and Refresh Token Request URLs all use a standard Salesforce domain, based on prod vs sandbox, which the users selects when configuring their account.

The Test Request that we've configure though, uses the `instance_url`, which is returned from the Access Token Request.

I am able to successfully authenticate with my Salesforce account.

@DilanAthukorala can you please send a screenshot of where you're getting stuck?

@aturow to confirm, is this an issue for you?

**Comment by DilanAthukorala** [comment]
Hi @dannyroosevelt
To update on my previous observation, I tried to recreate the issue again with a new Salesforce account and it was successful. However when I tried to reconnect to the same account again I observed the same issue as previous where the app is unable to connect to the url during oauth flow. And also observed that the URL is a bit different from my salesforce instance url. 
![image](https://user-images.githubusercontent.com/102980146/195184517-54c5ca6f-f1b9-4f91-a444-403047922c4f.png)
![image](https://user-images.githubusercontent.com/102980146/195184777-096ad68f-c869-41d9-a2fe-87b10e27abd0.png)

so this is only observed when trying to connect to the same account again.
